src/ngraph/frontend/onnx_import/op/hardmax.cpp

```diff
@@ -35,16 +36,25 @@ namespace ngraph
                 {
                     const auto input = node.get_ng_inputs().at(0);
                     const auto& input_shape = input->get_output_partial_shape(0);
-                    const auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
 
-                    const auto normalized_axis =
-                        ngraph::normalize_axis(node.get_description(), axis, input_shape.rank());
+                    auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
+                    if (input_shape.rank().is_static())
+                    {
+                        axis = ngraph::normalize_axis(
+                            node.get_description(), axis, input_shape.rank());
+                    }
```

Comment:
Can this `if` be performed inside `ngraph::normalize_axis` helper? We could return axis as is when rank is dynamic.

Reply:
`normalize_axis` throw exception if `rank` is `dynamic` and `axis<0`. In this case I think it is good to normalize if possible, if not: https://github.com/NervanaSystems/ngraph/blob/929a20e72dfb06b9a4cd64cdd266910333e304f3/src/ngraph/builder/reshape.cpp#L273 should handle it

Reply:
Since most of the operators support a negative axis(in the spec) It should be safe to return not normalized axis but it will mess up the logical flow.